Robot_Adapter/CRUD/Read/Loads/Loads.cs

```diff
@@ -259,6 +260,71 @@ private List<ILoad> ReadGeometricalLineLoads(List<string> ids = null)
+            List<ILoad> bhomLoads = new List<ILoad>();
+            Dictionary<int, Panel> bhomPanel = ReadPanels().ToDictionary(x => System.Convert.ToInt32(x.CustomData[AdapterIdName]));
+            Dictionary<string, Loadcase> bhomLoadCases = new Dictionary<string, Loadcase>();
+            List<Loadcase> lCases = ReadLoadCase();
+            for (int i = 0; i < lCases.Count; i++)
+            {
+                if (bhomLoadCases.ContainsKey(lCases[i].Name) == false)
+                    bhomLoadCases.Add(lCases[i].Name, lCases[i]);
+            }
+
+            //Dictionary<string, Loadcase> bhomLoadCases = ReadLoadCase().ToDictionary(x => x.Name);
+            IRobotCaseCollection loadCollection = m_RobotApplication.Project.Structure.Cases.GetAll();
+
+            for (int i = 1; i <= loadCollection.Count; i++)
+            {
+                IRobotCase lCase = loadCollection.Get(i) as IRobotCase;
+                if (lCase.Type == IRobotCaseType.I_CT_SIMPLE)
+                {
+                    IRobotSimpleCase sCase = lCase as IRobotSimpleCase;
+                    if (bhomLoadCases.ContainsKey(sCase.Name))
+                    {
+                        for (int j = 1; j <= sCase.Records.Count; j++)
+                        {
+                            IRobotLoadRecord loadRecord = sCase.Records.Get(j);
+                            List<int> elementIds = Convert.FromRobotSelectionString(loadRecord.Objects.ToText());
+                            List<Panel> objects = new List<Panel>();
+                            for (int k = 0; k < elementIds.Count; k++)
+                            {
+                                if (bhomPanel.ContainsKey(elementIds[k]))
+                                    objects.Add(bhomPanel[elementIds[k]]);
+                            }
+
+                            switch (loadRecord.Type)
+                            {
+                                case IRobotLoadRecordType.I_LRT_UNIFORM:
+                                    double fx = loadRecord.GetValue((short)IRobotUniformRecordValues.I_URV_PX);
+                                    double fy = loadRecord.GetValue((short)IRobotUniformRecordValues.I_URV_PY);
+                                    double fz = loadRecord.GetValue((short)IRobotUniformRecordValues.I_URV_PZ);
+                                    double ls = loadRecord.GetValue((short)IRobotUniformRecordValues.I_URV_LOCAL_SYSTEM);
+                                    double pj = loadRecord.GetValue((short)IRobotUniformRecordValues.I_URV_PROJECTED);
+
+
+                                    AreaUniformlyDistributedLoad PanelPressure = new AreaUniformlyDistributedLoad
+                                    {
+                                        Pressure = new Vector { X = fx, Y = fy, Z = fz },
+                                        Loadcase = bhomLoadCases[sCase.Name],
+                                        Objects = BH.Engine.Base.Create.BHoMGroup(objects) as BHoMGroup<IAreaElement>,
```

Comment:
Not exactly sure of why this doesn't work, but it doesn't
think this may be a more functioning way to write it:
`Objects = new BHoMGroup<IAreaElement>() { Elements = objects.ToList<IAreaElement>() },`
but would advise testing and asking @IsakNaslundBh

Reply:
good catch, I think the solution you proposed will work. 

First would not work cause a `BHoMGroup<Panel>` is not a `BHoMGroup<IAreaElement>`. (same as a `List<int>` is not a `List<object>`). Generics are fun!